### PR TITLE
feat(compiler): add support for super constructor

### PR DESCRIPTION
**Pull request by hasanaburayyan**
Adds support for calling super()  constructor within derived class initializer.

Closes: https://github.com/winglang/wing/issues/1123

## MISC:
- fixed latent bug where inflight client super constructor calls were not wrapping args as an object
- some minor refactors and cleanups

## Checklist

- [x] Title matches [Winglang's style guide](https://docs.winglang.io/contributors/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [x] Tests added (always)
- [x] Docs updated (only required for features)
- [x] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: examples/tests/invalid/class.w [hunk 1]
```diff
@@ -78,3 +78,83 @@ class C8 extends S1 {
...
+//  ^^^^^^^^ Expected 1 positional argument(s) but got 0
+    this.someStr = someStr;
+  }
+}
+
+class Four extends One {
+  someStr: str;
+  init(someNum: num, someStr: str) {
+    super(someNum, someStr);
+//  ^^^^^^^^^^^^^^^^^^^^^^^^ Expected 1 arguments but got 2
```
Comment (eladb) [inline examples/tests/invalid/class.w:142]: I suspect this will also work for keyword arguments, correct?

Comment (hasanaburayyan) [reply examples/tests/invalid/class.w:142]: I think it should since the type checking is the same but I forgot to make an example. Ill add now

Comment (hasanaburayyan) [reply examples/tests/invalid/class.w:142]: Looks like theres an issue with keyword args, looking into it.

File: tools/hangar/__snapshots__/test_corpus/valid/resource_captures_globals.w_compile_tf-aws.md [hunk 2]
```diff
@@ -91,10 +88,8 @@ module.exports = function({ globalBucket, globalStr, globalBool, globalNum, glob
       this.localTopic = localTopic;
     }
     async $inflight_init()  {
-      const __parent_this = this;
```
Comment (eladb) [inline tools/hangar/__snapshots__/test_corpus/valid/resource_captures_globals.w_compile_tf-aws.md:94]: you just saved lots of trees in the amazon forests by not emitting these redundant statements! thank you!

**Review by eladb — APPROVED** [review]

**Pull request merged**